examples/reqwest-response/src/main.rs

```diff
@@ -4,80 +4,93 @@
-//         .layer(TraceLayer::new_for_http().on_body_chunk(
-//             |chunk: &Bytes, _latency: Duration, _span: &Span| {
-//                 tracing::debug!("streaming {} bytes", chunk.len());
-//             },
-//         ))
-//         .with_state(client);
+    let app = Router::new()
+        .route("/", get(proxy_via_reqwest))
+        .route("/stream", get(stream_some_data))
+        // Add some logging so we can see the streams going through
+        .layer(TraceLayer::new_for_http().on_body_chunk(
+            |chunk: &Bytes, _latency: Duration, _span: &Span| {
+                tracing::debug!("streaming {} bytes", chunk.len());
+            },
+        ))
+        .with_state(client);
 
-//     let listener = tokio::net::TcpListener::bind("127.0.0.1:3000")
-//         .await
-//         .unwrap();
-//     tracing::debug!("listening on {}", listener.local_addr().unwrap());
-//     axum::serve(listener, app).await.unwrap();
-// }
+    let listener = tokio::net::TcpListener::bind("127.0.0.1:3000")
+        .await
+        .unwrap();
+    tracing::debug!("listening on {}", listener.local_addr().unwrap());
+    axum::serve(listener, app).await.unwrap();
+}
 
-// async fn proxy_via_reqwest(State(client): State<Client>) -> Response {
-//     let reqwest_response = match client.get("http://127.0.0.1:3000/stream").send().await {
-//         Ok(res) => res,
-//         Err(err) => {
-//             tracing::error!(%err, "request failed");
-//             return StatusCode::BAD_GATEWAY.into_response();
-//         }
-//     };
+async fn proxy_via_reqwest(State(client): State<Client>) -> Response {
+    let reqwest_response = match client.get("http://127.0.0.1:3000/stream").send().await {
+        Ok(res) => res,
+        Err(err) => {
+            tracing::error!(%err, "request failed");
+            return (StatusCode::BAD_REQUEST, Body::empty()).into_response();
+        }
+    };
 
-//     let mut response_builder = Response::builder().status(reqwest_response.status());
+    let mut response_builder = Response::builder().status(reqwest_response.status().as_u16());
 
-//     // This unwrap is fine because we haven't insert any headers yet so there can't be any invalid
-//     // headers
-//     *response_builder.headers_mut().unwrap() = reqwest_response.headers().clone();
+    // This unwrap is fine because we haven't insert any headers yet so there can't be any invalid
+    // headers
+    let headers = response_builder.headers_mut().unwrap();
+    for (header_name, header_value) in reqwest_response.headers().clone() {
+        headers.insert(
+            header_name
+                .to_owned()
+                .unwrap()
+                .as_str()
+                .parse::<HeaderName>()
+                .unwrap(),
+            header_value
+                .to_str()
+                .unwrap()
+                .parse::<HeaderValue>()
+                .unwrap(),
+        );
+    }
```

Comment:
It should be possible to convert the headers like this https://github.com/tokio-rs/axum/blob/main/axum/src/extract/multipart.rs#L98-L103. Thats a bit more efficient and doesn't require utf-8 encoded headers.